query-planner-js/src/__tests__/genTestSupergraph.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Generates a test supergraph, used in `supergraphBackwardCompatiblity.test.ts`, for the version passed in argument.
+ */
+
+import { generateTestSupergraph } from "./supergraphBackwardCompatibility.test";
+
+
+// This is called through ts-node so argv[0] is ts-node itself, argv[1] is this script file name, and argv[2] is thus our argument.
+const version = process.argv[2];
+if (!version) {
+  console.error('Missing mandatory version argument.');
+  process.exit(1);
+}
+
+// Ensure we're not given a completely broken argument. This also reject any `-alpha` or `-rc` versions on purpose: no
+// point in keeping test supergraphs for those versions around.
+if (!(/[0-9]+\.[0-9]+(\.[0-9]+)?/.test(version))) {
```

Comment:
We don't actually check that the file we are trying to generate is the branch that we're currently on. Probably nbd, but I thought I'd call it out as a potential footgun.

Reply:
Yeah; I wasn't actually sure how to do something useful here without being complicated. In theory, I wanted to use the current branch version instead of asking for a version manually, but as far as i can tell, we only bump versions just before releases, so the version of the "current branch" is usually the version of "whatever was the last release", and so it's not exactly the version we'd want for this.
I figured passing the version manually might end up less frustrating that something that tries to be automatic but does the wrong thing, but I'm open to suggestions (maybe someday we get this integrated to the release process...). Anyway, I'll note that we still need to manually commit the result of this script, which give more change to double check for mistakes.